vnext/Microsoft.ReactNative/Modules/DeviceInfoModule.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "DeviceInfoModule.h"
+#include <IReactDispatcher.h>
+#include <winrt/Windows.UI.Core.h>
+#include <winrt/Windows.UI.ViewManagement.h>
+
+namespace Microsoft::ReactNative {
+
+DeviceInfoHolder::DeviceInfoHolder() {
+  updateDeviceInfo();
+}
+
+void DeviceInfoHolder::InitDeviceInfoHolder(
+    const winrt::Microsoft::ReactNative::ReactPropertyBag &propertyBag) noexcept {
+  auto deviceInfoHolder = std::make_shared<DeviceInfoHolder>();
+  deviceInfoHolder->updateDeviceInfo();
+
+  propertyBag.Set(
+      winrt::Microsoft::ReactNative::ReactPropertyId<
+          winrt::Microsoft::ReactNative::ReactNonAbiValue<std::shared_ptr<DeviceInfoHolder>>>(
+          L"DeviceInfo", L"DeviceInfoHolder"),
+      std::move(deviceInfoHolder));
+
+  auto const &displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
+  auto const &window = xaml::Window::Current().CoreWindow();
+
+  deviceInfoHolder->m_sizeChangedRevoker =
+      window.SizeChanged(winrt::auto_revoke, [weakHolder = std::weak_ptr(deviceInfoHolder)](auto &&, auto &&) {
+        if (auto strongHolder = weakHolder.lock()) {
+          strongHolder->updateDeviceInfo();
+        }
+      });
+
+  deviceInfoHolder->m_dpiChangedRevoker = displayInfo.DpiChanged(
+      winrt::auto_revoke, [weakHolder = std::weak_ptr(deviceInfoHolder)](const auto &, const auto &) {
+        if (auto strongHolder = weakHolder.lock()) {
+          strongHolder->updateDeviceInfo();
+        }
+      });
+}
+
+void DeviceInfoHolder::notifyChanged() noexcept {
+  if (m_notifyCallback) {
+    m_notifyCallback(getDimensions());
+  }
+}
+
+React::JSValueObject DeviceInfoHolder::GetDimensions(
+    const winrt::Microsoft::ReactNative::ReactPropertyBag &propertyBag) noexcept {
+  auto holder = propertyBag.Get(winrt::Microsoft::ReactNative::ReactPropertyId<
+                                winrt::Microsoft::ReactNative::ReactNonAbiValue<std::shared_ptr<DeviceInfoHolder>>>(
+      L"DeviceInfo", L"DeviceInfoHolder"));
```

Comment:
>L"DeviceInfo", L"DeviceInfoHolder" [](start = 6, length = 34)

Consider using a static variable instead of duplicating it. To avoid the DLL load slow down, we could return the PropertyId from a static method that uses a static variable. It will create it on-demand.
Also, you may consider to use the 'React' namespace to make the code shorter. It is an alias for the 'winrt::Microsoft::ReactNative' namespace.